### DISCUSSION TITLE
Unicode code point via number pad?

### DISCUSSION THREAD

**Discussion by malonn23** [tsujan/FeatherPad#786] [Q&A] [answered]
I noticed that Unicode input does not work via the numpad.  Is this a code limitation or a configuration thing?  Just curious.

**Comment by tsujan** [comment]
FeatherPad supports what Qt does. If, at some point, Qt supports Unicode input with numpad, it'll show up in FeatherPad automatically.

**Reply by malonn23** [reply]
Thanks, @tsujan.  I'll have to dig into the source and Google to see if there is a need for another package.

**Reply by tsujan** [reply]
I might add codes to FeatherPad for supporting it in its text view — I think it'll be a good feature — but other text entries (like line-edits) depend on Qt itself.

**Reply by malonn23** [reply]
No problem, brother.  I admit that I have not looked at the source (don't know much C++), but I do know (as do you, apparently) that numpad support is present in Qt.  It may require an [QKeyEvent](https://doc.qt.io/qt-6/qkeyevent.html), i.e. it is not present in the QLineEdit widget (without sub-classing it, of course) by default.  Whatever you decide to do, I love your application.  My favorite text editor to date.

**Reply by tsujan** [reply]
With Qt and C++ you can do many things :) Actually, FeatherPad's code is full of examples, some of which work around Qt bugs. The question is, "will it be worth the additional code?" I think, in this case, it might. But I need to get my hands dirty to know for sure.

**Reply by tsujan** [reply, marked as answer]
OK, this is a relatively complete answer (which I found years ago but forgot later):

To support unicode input with all Qt apps, the user needs to install ibus, (auto-)start `ibus-daemon`, and set the environment variable `QT_IM_MODULE` to `ibus`. Then all Qt apps accept `Ctrl+Shift+U` + unicode + `Space/Enter`.

GTK supports `Ctrl+Shift+U` without needing ibus.

**Reply by tsujan** [reply]
BTW, the numbers can be written in any way; number pad isn't important.
